dot/sync/chain_sync.go

```diff
@@ -503,12 +516,38 @@ func (cs *chainSync) getTarget() *big.Int {
 		return big.NewInt(2<<32 - 1)
 	}
 
+	// we are going to sort the data and remove the outliers then we will return the avg of all the valid elements
+	intArr := make([]interface{}, len(cs.peerState))
 	for _, ps := range cs.peerState {
-		sum = big.NewInt(0).Add(sum, ps.number)
+		intArr = append(intArr, *ps.number)
+	}
+	//now sort the array
+	sort.Sort(BigIntSorter(intArr))
```

Comment:
is the `BigIntSorter` needed? you can just use `sort.Slice` and define the `Less()` function: https://pkg.go.dev/sort#Slice

Reply:
```suggestion
	sort.Slice(cs.peerState, func(i, j) bool {
	    cs.peerState[i].number < cs.peerState[j].number
	 }
```

Reply:
@noot sort.Slice is a fantastic idea. However, cs.peerState is a map. Iterating over a map in Golang doesn't ensure the order of the elements. Hence, we need to get the values from the cs.peerState first then apply the less fn.

Reply:
done ✔